### ISSUE TITLE
Image file outside workspace not refreshing on file update

### ISSUE THREAD

**Issue by renkun-ken** [microsoft/vscode#143291]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.64.2, 1.65.0-insider
- OS Version: macOS 12.2.1, Ubuntu 20.04

https://user-images.githubusercontent.com/4662568/154522478-dcecbb3d-f925-4b24-aec9-9cad53da802c.mp4

Steps to Reproduce:

```bash
FILE=$(mktemp).png
curl https://via.placeholder.com/500 -o $FILE
code $FILE
curl https://via.placeholder.com/400 -o $FILE
curl https://via.placeholder.com/300 -o $FILE
```

When the PNG file is open in the editor, it is properly displayed. But when the file is updated (replaced), the image does not refresh.

If the file is a text file, then the updating works normally. It could be reproduced with the following:

```bash
FILE=$(mktemp).txt
echo "hello" >> $FILE
code $FILE
echo "hello" >> $FILE
echo "hello" >> $FILE
```

**Comment by bpasero** [comment]
What editor is used to display the image?

/gifPlease

**Comment by bpasero** [comment]
@renkun-ken to figure out if this is a missing file event or an issue with the image editor can you:
* run `code --verbose`
* from the window open developer tools (from help menu) and go to Console
* type in a filter to reduce the log spam: `parcel`
* do the change to the image and see if an event is logged

![image](https://user-images.githubusercontent.com/900690/154525080-ac412ce3-b738-4ee6-8db8-bc20160314ed.png)

**Comment by renkun-ken** [comment]
The log looks like the following:

<img width="1235" alt="image" src="https://user-images.githubusercontent.com/4662568/154527775-4664bd3a-f6b3-46b5-b45a-f1b333c872b6.png">

**Comment by bpasero** [comment]
@renkun-ken it seems to me this file is not inside the workspace but outside the workspace, is that true? 

Maybe ELI5 how to set this up from scratch, I was not able to reproduce so far, but the file was inside the workspace.

**Comment by bpasero** [comment]
Oh sorry nevermind, your steps are attached....

**Comment by renkun-ken** [comment]
Yes, the file is outside workspace and in a temp folder. I tried creating the file at home folder but not in workspace, it does not refresh too. Looks like any image file outside workspace won't refresh.

**Comment by bpasero** [comment]
**Steps:**
* from integrated terminal of an empty window
* run these steps:

```
FILE=$(mktemp).png
curl https://via.placeholder.com/500 -o $FILE
code $FILE
curl https://via.placeholder.com/400 -o $FILE
curl https://via.placeholder.com/300 -o $FILE
```

@mjbvz reproduces in the image preview editor that I think you added

@Tyriar also reproduces in Luna paint.

**Comment by antheas** [comment]
Since you dragged me into this

```bash
mkdir -p tst/workspace tst/nonworkspace
code /tst/workspace
```

Then, in the code window
```bash
FILE=../nonworkspace/img.png
curl https://via.placeholder.com/500 -o $FILE
code $FILE
curl https://via.placeholder.com/400 -o $FILE
curl https://via.placeholder.com/300 -o $FILE
```

The preview doesn't update. However, with text it does work. 

So it doesn't have to do with the tmp directory and is probably what @bpasero says.

The image does update when the workplace is soft-linked, however, so my issue is different.

**Comment by Tyriar** [comment]
@bpasero I reported this a few months ago in https://github.com/microsoft/vscode/issues/137574 but you couldn't repro it with my steps 🤷. I haven't done any follow up since creating that issue. This is how I create the watcher in Luna Paint:

```ts
const watcher = workspace.createFileSystemWatcher(this._fileWatcherUri.fsPath);
```

Looks about as explicit as you can get because it's a file not a directory. Is the problem related to not using `RelativePattern` like in [the example](https://code.visualstudio.com/updates/v1_64#_vscodeworkspacecreatefilesystemwatcher-now-supports-any-path)?

**Comment by bpasero** [comment]
Yeah, meanwhile I have put a ton of docs to that method to make this clear, best is to put in a `Uri`, any `string` is interpreted as glob pattern.

**Comment by mjbvz** [comment]
Thanks @bpasero!

I pushed a fix for the image preview. In my testing, watching images outside of the workspace root now works as expected

**Comment by renkun-ken** [comment]
Thanks for the fix!

**Comment by Tyriar** [comment]
Thanks, will make a change this version

**Comment by TylerLeonhardt** [comment]
Doesn't seem to be fixed. I used Remote WSL for this one.

![Recording 2022-02-25 at 12 56 47](https://user-images.githubusercontent.com/2644648/155801833-3cffb69c-f18b-433a-97e4-460530d7a42f.gif)

**Comment by mjbvz** [comment]
Opened new issue for Remote WSL. Closing again for verification in non-remote cases
